### PR TITLE
Update import-boss restrictions - pkg/kubectl

### LINKED ISSUE #68201: Upate import-boss restrictions
`hack/verify-import-boss.sh` is a tool we use to reduce technical debt around rampant interdepencies between packages in the main k/k repository.

The tool uses a file `.import-restrictions` in some directories that verify-import-boss.sh then uses to check. The syntax for the file is here:
https://github.com/kubernetes/code-generator/blob/master/cmd/import-boss/main.go#L32-L52

We currently have the following `.import-restrictions` in the main k/k repository:
```
[dims@dims-mac 19:00] ~/go/src/k8s.io/kubernetes ⟩ find  . -name .import-res*
./cmd/kubeadm/.import-restrictions
./pkg/.import-restrictions
./pkg/api/testing/.import-restrictions
./pkg/client/clientset_generated/.import-restrictions
./pkg/controller/.import-restrictions
./pkg/kubectl/.import-restrictions
./pkg/kubemark/.import-restrictions
./pkg/printers/.import-restrictions
./pkg/printers/internalversion/.import-restrictions
./staging/src/k8s.io/apiserver/.import-restrictions
./staging/src/k8s.io/code-generator/cmd/lister-gen/.import-restrictions
```
What needs to be done?
- Open up the `.import-restrictions` file, check if `AllowedPrefixes` under different `Rules` are correct. 
  - The packages listed are still being used in the code under the pacakge
  - If there are outside packages that are used, but not listed under `AllowedPrefixes` then add those packages
- Possibly add new .import-restrictions for different directories under `/staging/src/k8s.io/` so we can be sure each directory under that is self-contained.
- If the .import-restrictions looks correct, then leave a comment in this issue so others don't duplicate the work. If you want to work on a specific file, leave a note so others don't spend their time on the same work.

Prior art (see how we did this in earlier PRs):
- https://github.com/kubernetes/kubernetes/pull/35148
- https://github.com/kubernetes/kubernetes/pull/53458
- https://github.com/kubernetes/kubernetes/pull/63740

/good-first-issue

### PR DESCRIPTION
**Pull request by saravanan30erd**
**What this PR does / why we need it**:
Update import-boss restrictions - pkg/kubectl

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Refer #68201 

**Special notes for your reviewer**:
```
$ hack/verify-import-boss.sh
+++ [0904 22:58:14] Building go targets for darwin/amd64:
    vendor/k8s.io/code-generator/cmd/import-boss

```

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by saravanan30erd** [comment]
@dims please correct me if I missed something.

File: pkg/kubectl/.import-restrictions [hunk 1]
```diff
@@ -1,151 +1,130 @@
...
-                "k8s.io/kubernetes/pkg/util/nsenter",
-                "k8s.io/kubernetes/pkg/util/parsers",
-                "k8s.io/kubernetes/pkg/util/slice",
-                "k8s.io/kubernetes/pkg/util/taints",
-                "k8s.io/kubernetes/pkg/version",
-                "k8s.io/kubernetes/pkg/version/prometheus",
-                "k8s.io/kubernetes/pkg/volume",
-                "k8s.io/kubernetes/pkg/volume/util",
-                "k8s.io/utils/pointer"
+			"k8s.io/kubernetes/pkg/api",
```
Comment (neolit123) [inline pkg/kubectl/.import-restrictions]: hm, any idea why the new diff uses tabs instead of spaces for the indentation?

**Comment by neolit123** [comment]
@saravanan30erd 
thanks for the PR!
it looks like you can squash the commits into a single one, as they touch the same file and make related changes.

File: pkg/kubectl/.import-restrictions (hunk 1 above)
Comment (saravanan30erd) [reply pkg/kubectl/.import-restrictions]: fixed now.

**Comment by saravanan30erd** [comment]
@neolit123 Done 👍 

```
$ hack/verify-import-boss.sh
+++ [0905 09:09:01] Building go targets for darwin/amd64:
    vendor/k8s.io/code-generator/cmd/import-boss

```

**Comment by neolit123** [comment]
SGTM, thanks for the update.

**Comment by idealhack** [comment]
@saravanan30erd  Please use `Ref #68201` instead of `Fixes #68201` to avoid closing that issue.

**Review by soltysh — APPROVED** [review]
/approve 
but I'll let @dims have final saying

**Comment by saravanan30erd** [comment]
@dims Could you provide any feedback for this?

**Comment by dims** [comment]
@saravanan30erd Ack. will do once we get past the 1.12 release freeze (a bit tied up at the moment)

File: pkg/kubectl/.import-restrictions [hunk 2]
```diff
@@ -1,151 +1,130 @@
...
+        "k8s.io/kubernetes/pkg/scheduler/api",
+        "k8s.io/kubernetes/pkg/scheduler/cache",
+        "k8s.io/kubernetes/pkg/scheduler/util",
+        "k8s.io/kubernetes/pkg/security/apparmor",
+        "k8s.io/kubernetes/pkg/util/labels",
+        "k8s.io/kubernetes/pkg/util/node",
+        "k8s.io/kubernetes/pkg/util/slice",
+        "k8s.io/kubernetes/pkg/features",
+        "k8s.io/kubernetes/pkg/registry/rbac/validation",
+        "k8s.io/kubernetes/pkg/scheduler/algorithm",
```
Comment (PrasadG193) [inline pkg/kubectl/.import-restrictions]: Hi @saravanan30erd,
Just trying to understand, there are lots of packages like -
```
k8s.io/kubernetes/pkg/scheduler/*
k8s.io/kubernetes/pkg/util/node
k8s.io/kubernetes/pkg/util/slice
k8s.io/kubernetes/pkg/features
k8s.io/kubernetes/pkg/registry/rbac/validation, etc
```
that are listed under `AllowedPrefixes` but I don't think they are imported anywhere under  `pkg/kubectl`, any specific reason why they are kept or am I missing something?

Comment (PrasadG193) [reply pkg/kubectl/.import-restrictions]: Aah, that was due to transitive dependencies, that was obvious. Sorry for the confusion.

**Pull request closed without merging**